src/main/java/org/teamresistance/frc/OpticalFlow.java

```diff
@@ -0,0 +1,165 @@
+package org.teamresistance.frc;
+
+import edu.wpi.first.wpilibj.SPI;
+import edu.wpi.first.wpilibj.SPI.Port;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import org.strongback.command.Command;
+
+import java.util.function.Supplier;
+
+/**
+ * Created by Joseph for testing the Optical Flow
+ *
+ * Registers
+ * 0x00  -   Product ID
+ * 0x01  -   Revision ID
+ * 0x02  -   Motion
+ * 0x03  -   Delta x
+ * 0x04  -   Delta y
+ * 0x05  -   Squall
+ *
+ */
+public class OpticalFlow {
+
+  private  final SPI spi;
+  private  final byte[] dataReceived;
+  private  final byte[] register = new byte[] {0};
+
+  //Amount of change since last read
+  private long raw_dx = 0;
+  private long raw_dy = 0;
+
+  //Summation of raw_dx/y over time
+  private long tot_dx = 0;
+  private long tot_dy = 0;
+
+
+  //Convert for Decimal Values
+  //Pixel to Feet conversion factor
+  private long X_Left_Per_Ft = 590;
+  private long X_Right_Per_Ft = 575;
+  private long Y_Fwd_Per_Ft = 595;
+  private long Y_Rev_Per_Ft = 595;
+
+  //Distance Covered
+  private long dx = 0;
+  private long dy = 0;
+
+  //Vars for Franks code
+//  double xFacFt = 800;
+//  double yFacFt = 800;
+
+  public OpticalFlow() {
+    spi = new SPI(Port.kOnboardCS0);    //Finds the OF on the SPI ports
+    spi.setChipSelectActiveLow();
+    spi.setClockActiveHigh();
+    spi.setClockRate(500000);
+
+    dataReceived = new byte[1];
+    for(int i = 0; i < dataReceived.length; i++) {
+      dataReceived[i] = 0;
+    }
+  }
+
+  public void init() {
+    int motionRegister = readRegister((byte)2);
+    SmartDashboard.putNumber("Motion Register", motionRegister);
+
```

Comment:
I see you had removed the `readRegister` block that you originally had here. I'm referring to this thing:

```java
if((motionRegister & 0x80) != 0) {
      readRegister((byte)3);
      readRegister((byte)4);
}
```

Did testing show this block of code wasn't needed?